src/deprecated/deprecated.js

```diff
@@ -514,6 +515,42 @@ Object.keys(deprecatedChunks).forEach((chunkName) => {
     });
 });
 
+/**
+ * Helper function to ensure a bit of backwards compatibility.
+ *
+ * @example
+ * toLitArgs('litShaderArgs.sheen.specularity'); // Result: 'litArgs_sheen_specularity'
+ * @param {string} src - The shader source which may generate shader errors.
+ * @returns {string} The backwards compatible shader source.
+ */
+function compatibilityForLitArgs(src) {
+    if (src.includes('litShaderArgs')) {
+        src = src.replace(/litShaderArgs([\.a-zA-Z]+)+/g, (a, b) => {
+            const newSource = 'litArgs' + b.replace(/\./g, '_');
+            Debug.deprecated(`Nested struct property access is deprecated, because it's crashing some devices. Please update your custom chunks manually. In particular ${a} should be ${newSource} now.`);
+            return newSource;
+        });
+    }
+    return src;
+}
+
+/**
+ * Add more backwards compatibility functions as needed.
+ *
+ * @param {string} src - The fragment shader source which may generate shader errors.
+ * @returns {string} The backwards compatible fragment shader source.
+ */
+function compatibilityForFragmentShader(src) {
+    src = compatibilityForLitArgs(src);
+    return src;
+}
+
+const original_LitShader_generateFragmentShader = LitShader.prototype.generateFragmentShader;
```

Comment:
To be honest, I'm not super keen on the engine monkey patching its internal functions, as its harder to find this is taking place. I'd much preferred a cleaner approach such as this (using just some random naming)

in deprecated.js
```
LitShader.prototype.fixup = function () {
    // do the validation / fixing here
}
```

in lit-shader.js
```
generateFragmentShader() {

    // at some point
    if (this.fixup) {
        code = this.fixup(code);
    }
}
```

Reply:
Good idea, I wasn't too happy about the long comment explaining the situation over the method either. First I didn't understand what you meant by `ChunkBuilder.handleCompatibility`, but now I got it. I will rewrite it tomorrow.

Reply:
Note that the inside of the `generateFragmentShader` could also be inside Debug.call() to only be included in the debug build.

Reply:
Hi .. if you have some time to do this very soon it'd be awesome, otherwise I'm happy to take over if you're too busy .. it's the last PR we're like to merge before starting the engine release. Please let me know. @kungfooman

Reply:
Sorry, I just read it, I will fix it right away